api/types/authority.go

```diff
@@ -288,15 +288,66 @@ func (ca *CertAuthorityV2) SetSigningAlg(alg CertAuthoritySpecV2_SigningAlgType)
 	ca.Spec.SigningAlg = alg
 }
 
-func (ca *CertAuthorityV2) GetActiveKeys() CAKeySet { return ca.Spec.ActiveKeys }
+func (ca *CertAuthorityV2) GetActiveKeys() CAKeySet {
+	haveNewCAKeys := len(ca.Spec.ActiveKeys.SSH) > 0 || len(ca.Spec.ActiveKeys.TLS) > 0 || len(ca.Spec.ActiveKeys.JWT) > 0
+	if haveNewCAKeys {
+		return ca.Spec.ActiveKeys
+	}
+	// fall back to old schema
+	var keySet CAKeySet
+	if len(ca.Spec.CheckingKeys) > 0 {
+		kp := &SSHKeyPair{
+			PrivateKeyType: PrivateKeyType_RAW,
+			PublicKey:      utils.CopyByteSlice(ca.Spec.CheckingKeys[0]),
+		}
+		if len(ca.Spec.SigningKeys) > 0 {
+			kp.PrivateKey = utils.CopyByteSlice(ca.Spec.SigningKeys[0])
+		}
+		keySet.SSH = []*SSHKeyPair{kp}
+	}
+	if len(ca.Spec.TLSKeyPairs) > 0 {
+		keySet.TLS = []*TLSKeyPair{ca.Spec.TLSKeyPairs[0].Clone()}
+	}
+	if len(ca.Spec.JWTKeyPairs) > 0 {
+		keySet.JWT = []*JWTKeyPair{ca.Spec.JWTKeyPairs[0].Clone()}
+	}
+	return keySet
```

Comment:
Can you move this logic to a function like `func (ca *CertAuthorityV2) getOldKeySet(index uint) CaKeySet` and use that in `GetActiveKeys` and `GetAdditionalTrustedKeys`?

Reply:
Good idea, done in https://github.com/gravitational/teleport/pull/7603/commits/8d83e6b5d57ab77a747a231e39e52c03e470c132